File: Первый курс/Первый семестр/Практикум на ЭВМ/Паскаль/11.pas
const
  N = 10;

type
  mas = array[1..N] of integer;

var
  a: mas;
  i: integer;

procedure findandreplace(var array1: mas);
var
  min, i: integer;
begin
  min := abs(a[1]);
  for i := 1 to N do
    if min > abs(a[i]) then
      min := abs(array1[i]);
  i := 1;
  while i <= N do
  begin
    array1[i] := min;
    i := i + 2;
  end;
end;

begin
  for i := 1 to N do
  begin
    write('Введите ', i, ' элемент ');
    readln(a[i])
  end;
  findandreplace(a);
  for i := 1 to N do
    write(a[i], ' ');
  writeln();
end.

File: Первый курс/Первый семестр/Практикум на ЭВМ/C++/11.cpp
#include <cmath>
#include <iostream>

using namespace std;

void find_and_replace(int *array, int size) {
    int min = abs(array[0]);
    for (int i = 0; i < size; i++)
        if (min > abs(array[i]))
            min = abs(array[i]);
    for (int i = 0; i < size; i += 2) {
        array[i] = min;
    }
}

int main() {
    int n;
    cout << "Введите размерность массива" << '\n';
    cin >> n;
    int a[n];
    for (int i = 0; i < n; i++) {
        cout << "Введите " << i + 1 << " элемент массива ";
        cin >> a[i];
    }
    find_and_replace(a, n);
    for (int i = 0; i < n; i++) {
        cout << a[i] << ' ';
    }
    cout << '\n';
    return 0;
}
